src/main/anovos/data_transformer/transformers.py

```diff
@@ -824,48 +838,55 @@ def cat_to_num_supervised(
                 .drop(*["1", "0"])
             )
 
-            if model_path == "NA":
-                model_path = "intermediate_data"
-
-            df_tmp.coalesce(1).write.csv(
-                model_path + "/cat_to_num_supervised/" + i,
-                header=True,
-                mode="overwrite",
-            )
-            df_tmp = spark.read.csv(
-                model_path + "/cat_to_num_supervised/" + i,
-                header=True,
-                inferSchema=True,
-            )
+            if save_model:
+                try:
+                    df_tmp.coalesce(1).write.csv(
+                        model_path + "/cat_to_num_supervised/" + i,
+                        header=True,
+                        mode="overwrite",
+                        ignoreLeadingWhiteSpace=False,
+                        ignoreTrailingWhiteSpace=False,
+                    )
+                    df_tmp = spark.read.csv(
+                        model_path + "/cat_to_num_supervised/" + i,
+                        header=True,
+                        inferSchema=True,
+                    )
+                except Exception as error:
+                    if skip_if_error:
+                        warnings.warn(
+                            "For optimization purpose, we recommend specifying a valid model_path value to save the intermediate data. Saving to the default path - '"
+                            + model_path
+                            + "/cat_to_num_supervised/"
+                            + i
+                            + "' faced an error."
+                        )
+                        save_model = False
+                    else:
+                        raise error
 
         if df_tmp.count() > 1:
-            odf_partial = odf_partial.join(df_tmp, i, "left_outer")
+            odf = odf.join(F.broadcast(df_tmp), i, "left_outer")
```

Comment:
Using `broadcast` to optimize the join method is a nice approach, but need to also take into account the fact that 
`spark.sql.autoBroadcastJoinThreshold` by default is only set at 10MB, and the maximum size of a broadcast table is 8GB. Since both of these thresholds we cannot change nor modify, using `broadcast` to join in the function might cause some scalability issues later, so I would recommend against it.